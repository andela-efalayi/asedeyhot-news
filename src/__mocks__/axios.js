import testdata from './dataMock.json';

const mockApiCall = {
  get() {
    return Promise.resolve(testdata);
  }
};

export default mockApiCall;
