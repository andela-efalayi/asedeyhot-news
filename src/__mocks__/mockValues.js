/**
 * Mock values for app components
 */
export const articleWithNoDescription = {
  author: 'BBC News',
  title: 'US submarine arrives in South Korea',
  description: undefined,
  url: 'http://www.bbc.co.uk/news/world-asia-39701481',
  urlToImage: 'http://image-url/image.jpg',
  publishedAt: '2017-04-25T02:59:53+00:00'
};

export const articleWithDescription = {
  author: 'BBC News',
  title: 'US submarine arrives in South Korea',
  description: 'It comes amid worries of a North Korean missile',
  url: 'http://www.bbc.co.uk/news/world-asia-39701481',
  urlToImage: 'http://image-url/image.jpg',
  publishedAt: '2017-04-25T02:59:53+00:00'
};

export const favouriteArticles = {
  title: {
    author: 'BBC News',
    description: 'The party says it would prioritise',
    url: 'http://www.bbc.co.uk/news/uk-politics-39698465'
  }
};

export const favouriteSources = {
  'Associated Press': 'http://bigstory.ap.org'
};

export const sortBarDetails = {
  currentSortOption: 'top',
  title: 'ABC News (AU)',
  sortOptions: ['top']
};

export const sourceWithLongDescription = {
  id: 'al-jazeera-english',
  name: 'Al Jazeera English',
  category: 'general',
  description: 'News, analysis from the Middle East and worldwide,' +
    'multimedia and interactives, opinions, documentaries, podcasts,' +
    'long reads and broadcast schedule.',
  url: 'http://www.aljazeera.com',
  sortBysAvailable: ['top', 'latest']
};

export const sourceWithShortDescription = {
  id: 'al-jazeera-english',
  name: 'Al Jazeera English',
  category: 'general',
  description: 'News, analysis from the Middle East',
  url: 'http://www.aljazeera.com',
  sortBysAvailable: ['top', 'latest']
};

export const user = {
  name: 'Username',
  imageUrl: 'https://dummyimage.com/600x400/000000/0011ff'
};
