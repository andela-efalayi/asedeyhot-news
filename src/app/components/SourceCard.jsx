import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { Card, CardActions, CardText } from 'material-ui/Card';
import swal from 'sweetalert2';
import IconButton from 'material-ui/IconButton';
import FavoriteBorder from 'material-ui/svg-icons/action/favorite-border';
import Favorite from 'material-ui/svg-icons/action/favorite';
import ZoomIn from 'material-ui/svg-icons/action/zoom-in';
import AppFavourites from '../actions/AppFavourites';

/**
 * @class SourceCard
 * @extends {Component}
 * @method saveToFavouriteSources
 * @method setCurrentSource
 * @param {object} props
 */

class SourceCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      favIcon: <FavoriteBorder />,
      tooltip: 'Add to favourite sources'
    };
    this.setCurrentSource = this.setCurrentSource.bind(this);
    this.saveToFavouriteSources = this.saveToFavouriteSources.bind(this);
  }

  /**
   * Saves a new favourite source to the database
   * @memberof SourceCard
   * @method saveToFavouriteSources
   * @return {void}
   */

  saveToFavouriteSources() {
    const source = this.props.source;
    AppFavourites.saveToFavouriteSources(source)
    .then((status) => {
      // Update view if successful
      this.setState({
        favIcon: <Favorite />,
        tooltip: status
      });
    })
    .catch((error) => {
      // Update view if successful
      swal(
        'Sorry',
         error,
        'error'
      );
    });
  }

  /**
   * Save a currently selected source to localStorage
   * @memberof SourceCard
   * @method setCurrentSource
   * @return {void}
   */

  setCurrentSource(source) {
    const sourceDetails = {
      id: source.id,
      name: source.name,
      sortOptions: source.sortBysAvailable
    };
    localStorage.setItem('currentSource', JSON.stringify(sourceDetails));
  }

  render() {
    const source = this.props.source;
    const setCurrentSource = () => this.setCurrentSource(source);
    const setChar = (word, limit, endChar) => {
      const newWord = word;
      if (newWord.length > limit) {
        return `${newWord.substring(0, limit) + endChar}`;
      }
      return word;
    };
    return (
      <Card className="card">
        <div className="title">
          <h3 onClick={setCurrentSource}>
            <Link to={{
              pathname: `/sources/${source.id}`
            }}>
          {source.name} </Link></h3>
          <h6>{source.category.toUpperCase()}</h6>
        </div>
        <CardText className="card-description">
          {setChar(this.props.source.description,
            120,
            '...'
          )}
        </CardText>
        <CardActions className="source-actions">
          <IconButton tooltip={this.state.tooltip} touch
            onClick={this.saveToFavouriteSources}
            tooltipPosition="bottom-right">
            {this.state.favIcon}
          </IconButton>
          <IconButton tooltip="View headlines" touch
            onClick={setCurrentSource}
            containerElement={<Link to={`/sources/${source.id}`}/>}
            tooltipPosition="bottom-right">
            <ZoomIn/>
          </IconButton>
        </CardActions>
      </Card>
    );
  }
}

SourceCard.propTypes = {
  source: PropTypes.object
};

export default SourceCard;
