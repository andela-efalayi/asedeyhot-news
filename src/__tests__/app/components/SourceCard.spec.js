import React from 'react';
import { shallow } from 'enzyme';
import SourceCard from '../../../app/components/SourceCard.jsx';
import * as mockValue from '../../../__mocks__/mockValues';

describe('SourceCard.jsx', () => {
  const wrapper = shallow(
    <SourceCard source={mockValue.sourceWithLongDescription}/>
  );
  const anotherWrapper = shallow(
    <SourceCard source={mockValue.sourceWithShortDescription}/>
  );

  describe('Component', () => {
    it('should match match snaphots', () => {
      expect(wrapper).toMatchSnapshot();
      expect(anotherWrapper).toMatchSnapshot();
    });
  });

  describe('Component with long source description', () => {
    it('should have a description truncated to 123 characters', () => {
      expect(wrapper.find('.card-description').childAt(0).node.length)
      .toEqual(123);
    });
  });

  describe('Component with short source description', () => {
    it('should not be truncated', () => {
      expect(anotherWrapper.find('.card-description').childAt(0).node.length)
      .toEqual(mockValue.sourceWithShortDescription.description.length);
    });
  });

  describe('.card-actions', () => {
    it('should trigger an onClick event when an icon is clicked', () => {
      wrapper.find('.source-actions').childAt(0).simulate('click');
      expect(wrapper.find('.source-actions').childAt(0)
      .node.props.onClick).toBeDefined();

      wrapper.find('.source-actions').childAt(1).simulate('click');
      expect(wrapper.find('.source-actions').childAt(1)
      .node.props.onClick).toBeDefined();
    });
  });
});
