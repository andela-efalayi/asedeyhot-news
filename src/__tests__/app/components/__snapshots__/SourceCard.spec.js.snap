// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceCard.jsx Component should match match snaphots 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card
    className="card"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
>
    <div
        className="title"
    >
        <h3
            onClick={[Function]}
        >
            <Link
                replace={false}
                to={
                    Object {
                        "pathname": "/sources/al-jazeera-english",
                      }
                }
            >
                Al Jazeera English
                 
            </Link>
        </h3>
        <h6>
            GENERAL
        </h6>
    </div>
    <CardText
        className="card-description"
    >
        News, analysis from the Middle East and worldwide,multimedia and interactives, opinions, documentaries, podcasts,long re...
    </CardText>
    <CardActions
        className="source-actions"
    >
        <IconButton
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            onClick={[Function]}
            tooltip="Add to favourite sources"
            tooltipPosition="bottom-right"
            touch={true}
        >
            <ActionFavoriteBorder />
        </IconButton>
        <IconButton
            containerElement={
                <Link
                    replace={false}
                    to="/sources/al-jazeera-english"
                />
            }
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            onClick={[Function]}
            tooltip="View headlines"
            tooltipPosition="bottom-right"
            touch={true}
        >
            <ActionZoomIn />
        </IconButton>
    </CardActions>
</Card>,
  "nodes": Array [
    <Card
      className="card"
      expandable={false}
      expanded={null}
      initiallyExpanded={false}
>
      <div
            className="title"
      >
            <h3
                  onClick={[Function]}
            >
                  <Link
                        replace={false}
                        to={
                              Object {
                                    "pathname": "/sources/al-jazeera-english",
                                  }
                        }
                  >
                        Al Jazeera English
                         
                  </Link>
            </h3>
            <h6>
                  GENERAL
            </h6>
      </div>
      <CardText
            className="card-description"
      >
            News, analysis from the Middle East and worldwide,multimedia and interactives, opinions, documentaries, podcasts,long re...
      </CardText>
      <CardActions
            className="source-actions"
      >
            <IconButton
                  disableTouchRipple={false}
                  disabled={false}
                  iconStyle={Object {}}
                  onClick={[Function]}
                  tooltip="Add to favourite sources"
                  tooltipPosition="bottom-right"
                  touch={true}
            >
                  <ActionFavoriteBorder />
            </IconButton>
            <IconButton
                  containerElement={
                        <Link
                              replace={false}
                              to="/sources/al-jazeera-english"
                        />
                  }
                  disableTouchRipple={false}
                  disabled={false}
                  iconStyle={Object {}}
                  onClick={[Function]}
                  tooltip="View headlines"
                  tooltipPosition="bottom-right"
                  touch={true}
            >
                  <ActionZoomIn />
            </IconButton>
      </CardActions>
</Card>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SourceCard
        source={
                Object {
                        "category": "general",
                        "description": "News, analysis from the Middle East and worldwide,multimedia and interactives, opinions, documentaries, podcasts,long reads and broadcast schedule.",
                        "id": "al-jazeera-english",
                        "name": "Al Jazeera English",
                        "sortBysAvailable": Array [
                          "top",
                          "latest",
                        ],
                        "url": "http://www.aljazeera.com",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SourceCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "source": Object {
            "category": "general",
            "description": "News, analysis from the Middle East and worldwide,multimedia and interactives, opinions, documentaries, podcasts,long reads and broadcast schedule.",
            "id": "al-jazeera-english",
            "name": "Al Jazeera English",
            "sortBysAvailable": Array [
              "top",
              "latest",
            ],
            "url": "http://www.aljazeera.com",
          },
        },
        "refs": Object {},
        "saveToFavouriteSources": [Function],
        "setCurrentSource": [Function],
        "state": Object {
          "favIcon": <ActionFavoriteBorder />,
          "tooltip": "Add to favourite sources",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card
          className="card"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
>
          <div
                    className="title"
          >
                    <h3
                              onClick={[Function]}
                    >
                              <Link
                                        replace={false}
                                        to={
                                                  Object {
                                                            "pathname": "/sources/al-jazeera-english",
                                                          }
                                        }
                              >
                                        Al Jazeera English
                                         
                              </Link>
                    </h3>
                    <h6>
                              GENERAL
                    </h6>
          </div>
          <CardText
                    className="card-description"
          >
                    News, analysis from the Middle East and worldwide,multimedia and interactives, opinions, documentaries, podcasts,long re...
          </CardText>
          <CardActions
                    className="source-actions"
          >
                    <IconButton
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="Add to favourite sources"
                              tooltipPosition="bottom-right"
                              touch={true}
                    >
                              <ActionFavoriteBorder />
                    </IconButton>
                    <IconButton
                              containerElement={
                                        <Link
                                                  replace={false}
                                                  to="/sources/al-jazeera-english"
                                        />
                              }
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="View headlines"
                              tooltipPosition="bottom-right"
                              touch={true}
                    >
                              <ActionZoomIn />
                    </IconButton>
          </CardActions>
</Card>,
        "_debugID": 2,
        "_renderedOutput": <Card
          className="card"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
>
          <div
                    className="title"
          >
                    <h3
                              onClick={[Function]}
                    >
                              <Link
                                        replace={false}
                                        to={
                                                  Object {
                                                            "pathname": "/sources/al-jazeera-english",
                                                          }
                                        }
                              >
                                        Al Jazeera English
                                         
                              </Link>
                    </h3>
                    <h6>
                              GENERAL
                    </h6>
          </div>
          <CardText
                    className="card-description"
          >
                    News, analysis from the Middle East and worldwide,multimedia and interactives, opinions, documentaries, podcasts,long re...
          </CardText>
          <CardActions
                    className="source-actions"
          >
                    <IconButton
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="Add to favourite sources"
                              tooltipPosition="bottom-right"
                              touch={true}
                    >
                              <ActionFavoriteBorder />
                    </IconButton>
                    <IconButton
                              containerElement={
                                        <Link
                                                  replace={false}
                                                  to="/sources/al-jazeera-english"
                                        />
                              }
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="View headlines"
                              tooltipPosition="bottom-right"
                              touch={true}
                    >
                              <ActionZoomIn />
                    </IconButton>
          </CardActions>
</Card>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SourceCard
    source={
        Object {
            "category": "general",
            "description": "News, analysis from the Middle East and worldwide,multimedia and interactives, opinions, documentaries, podcasts,long reads and broadcast schedule.",
            "id": "al-jazeera-english",
            "name": "Al Jazeera English",
            "sortBysAvailable": Array [
              "top",
              "latest",
            ],
            "url": "http://www.aljazeera.com",
          }
    }
/>,
}
`;

exports[`SourceCard.jsx Component should match match snaphots 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card
    className="card"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
>
    <div
        className="title"
    >
        <h3
            onClick={[Function]}
        >
            <Link
                replace={false}
                to={
                    Object {
                        "pathname": "/sources/al-jazeera-english",
                      }
                }
            >
                Al Jazeera English
                 
            </Link>
        </h3>
        <h6>
            GENERAL
        </h6>
    </div>
    <CardText
        className="card-description"
    >
        News, analysis from the Middle East
    </CardText>
    <CardActions
        className="source-actions"
    >
        <IconButton
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            onClick={[Function]}
            tooltip="Add to favourite sources"
            tooltipPosition="bottom-right"
            touch={true}
        >
            <ActionFavoriteBorder />
        </IconButton>
        <IconButton
            containerElement={
                <Link
                    replace={false}
                    to="/sources/al-jazeera-english"
                />
            }
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            onClick={[Function]}
            tooltip="View headlines"
            tooltipPosition="bottom-right"
            touch={true}
        >
            <ActionZoomIn />
        </IconButton>
    </CardActions>
</Card>,
  "nodes": Array [
    <Card
      className="card"
      expandable={false}
      expanded={null}
      initiallyExpanded={false}
>
      <div
            className="title"
      >
            <h3
                  onClick={[Function]}
            >
                  <Link
                        replace={false}
                        to={
                              Object {
                                    "pathname": "/sources/al-jazeera-english",
                                  }
                        }
                  >
                        Al Jazeera English
                         
                  </Link>
            </h3>
            <h6>
                  GENERAL
            </h6>
      </div>
      <CardText
            className="card-description"
      >
            News, analysis from the Middle East
      </CardText>
      <CardActions
            className="source-actions"
      >
            <IconButton
                  disableTouchRipple={false}
                  disabled={false}
                  iconStyle={Object {}}
                  onClick={[Function]}
                  tooltip="Add to favourite sources"
                  tooltipPosition="bottom-right"
                  touch={true}
            >
                  <ActionFavoriteBorder />
            </IconButton>
            <IconButton
                  containerElement={
                        <Link
                              replace={false}
                              to="/sources/al-jazeera-english"
                        />
                  }
                  disableTouchRipple={false}
                  disabled={false}
                  iconStyle={Object {}}
                  onClick={[Function]}
                  tooltip="View headlines"
                  tooltipPosition="bottom-right"
                  touch={true}
            >
                  <ActionZoomIn />
            </IconButton>
      </CardActions>
</Card>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SourceCard
        source={
                Object {
                        "category": "general",
                        "description": "News, analysis from the Middle East",
                        "id": "al-jazeera-english",
                        "name": "Al Jazeera English",
                        "sortBysAvailable": Array [
                          "top",
                          "latest",
                        ],
                        "url": "http://www.aljazeera.com",
                      }
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SourceCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "source": Object {
            "category": "general",
            "description": "News, analysis from the Middle East",
            "id": "al-jazeera-english",
            "name": "Al Jazeera English",
            "sortBysAvailable": Array [
              "top",
              "latest",
            ],
            "url": "http://www.aljazeera.com",
          },
        },
        "refs": Object {},
        "saveToFavouriteSources": [Function],
        "setCurrentSource": [Function],
        "state": Object {
          "favIcon": <ActionFavoriteBorder />,
          "tooltip": "Add to favourite sources",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card
          className="card"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
>
          <div
                    className="title"
          >
                    <h3
                              onClick={[Function]}
                    >
                              <Link
                                        replace={false}
                                        to={
                                                  Object {
                                                            "pathname": "/sources/al-jazeera-english",
                                                          }
                                        }
                              >
                                        Al Jazeera English
                                         
                              </Link>
                    </h3>
                    <h6>
                              GENERAL
                    </h6>
          </div>
          <CardText
                    className="card-description"
          >
                    News, analysis from the Middle East
          </CardText>
          <CardActions
                    className="source-actions"
          >
                    <IconButton
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="Add to favourite sources"
                              tooltipPosition="bottom-right"
                              touch={true}
                    >
                              <ActionFavoriteBorder />
                    </IconButton>
                    <IconButton
                              containerElement={
                                        <Link
                                                  replace={false}
                                                  to="/sources/al-jazeera-english"
                                        />
                              }
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="View headlines"
                              tooltipPosition="bottom-right"
                              touch={true}
                    >
                              <ActionZoomIn />
                    </IconButton>
          </CardActions>
</Card>,
        "_debugID": 4,
        "_renderedOutput": <Card
          className="card"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
>
          <div
                    className="title"
          >
                    <h3
                              onClick={[Function]}
                    >
                              <Link
                                        replace={false}
                                        to={
                                                  Object {
                                                            "pathname": "/sources/al-jazeera-english",
                                                          }
                                        }
                              >
                                        Al Jazeera English
                                         
                              </Link>
                    </h3>
                    <h6>
                              GENERAL
                    </h6>
          </div>
          <CardText
                    className="card-description"
          >
                    News, analysis from the Middle East
          </CardText>
          <CardActions
                    className="source-actions"
          >
                    <IconButton
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="Add to favourite sources"
                              tooltipPosition="bottom-right"
                              touch={true}
                    >
                              <ActionFavoriteBorder />
                    </IconButton>
                    <IconButton
                              containerElement={
                                        <Link
                                                  replace={false}
                                                  to="/sources/al-jazeera-english"
                                        />
                              }
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              onClick={[Function]}
                              tooltip="View headlines"
                              tooltipPosition="bottom-right"
                              touch={true}
                    >
                              <ActionZoomIn />
                    </IconButton>
          </CardActions>
</Card>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SourceCard
    source={
        Object {
            "category": "general",
            "description": "News, analysis from the Middle East",
            "id": "al-jazeera-english",
            "name": "Al Jazeera English",
            "sortBysAvailable": Array [
              "top",
              "latest",
            ],
            "url": "http://www.aljazeera.com",
          }
    }
/>,
}
`;
